tests/python/pants_test/util/test_objects.py

```diff
@@ -310,36 +310,47 @@ def test_class_construction_errors(self):
     with self.assertRaises(ValueError) as cm:
       class NonStrType(datatype([int])): pass
     expected_msg = (
-      "Type names and field names can only contain alphanumeric "
-      "characters and underscores: \"<type 'int'>\"")
+      "Type names and field names must be valid identifiers: \"<class 'int'>\""
+      if PY3 else
+      "Type names and field names can only contain alphanumeric characters and underscores: \"<type 'int'>\""
+    )
```

Comment:
A couple of error messages changed their wording between Py2 and Py3. The changes come from `TypeError`, and are not hardcoded on our end.

This should not impact any actual source code. The errors still raise in the same situations and the messages are essentially the same. 

@cosmicexplorer and I discussed via Slack.